pulsar-broker/src/main/java/org/apache/pulsar/broker/transaction/pendingack/impl/MLPendingAckStore.java

```diff
@@ -241,8 +241,8 @@ public void addComplete(Position position, ByteBuf entryData, Object ctx) {
                 completableFuture.complete(null);
 
                 if (!metadataPositions.isEmpty()) {
-                    PositionImpl firstPosition = metadataPositions.firstEntry().getKey();
-                    PositionImpl deletePosition = metadataPositions.firstEntry().getKey();
+                    PositionImpl firstPosition = PositionImpl.EARLIEST;
```

Comment:
I'm not sure why need `firstPosition`, maybe we can initialize the `deletePosition` as a null value, after checking, if the `deletePosition` is not null, then we can call the method `cursor.asyncMarkDelete`.

Reply:
I think you are right.

Reply:
They are equivalent